aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/operation/AWSS3StoragePathUploadFileOperation.kt

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amplifyframework.storage.s3.operation
+
+import com.amplifyframework.auth.AuthCredentialsProvider
+import com.amplifyframework.core.Amplify
+import com.amplifyframework.core.Consumer
+import com.amplifyframework.hub.HubChannel
+import com.amplifyframework.hub.HubEvent
+import com.amplifyframework.storage.ObjectMetadata
+import com.amplifyframework.storage.StorageChannelEventName
+import com.amplifyframework.storage.StorageException
+import com.amplifyframework.storage.TransferState
+import com.amplifyframework.storage.operation.StorageUploadFileOperation
+import com.amplifyframework.storage.result.StorageTransferProgress
+import com.amplifyframework.storage.result.StorageUploadFileResult
+import com.amplifyframework.storage.s3.ServerSideEncryption
+import com.amplifyframework.storage.s3.extensions.toS3ServiceKey
+import com.amplifyframework.storage.s3.request.AWSS3StoragePathUploadRequest
+import com.amplifyframework.storage.s3.service.StorageService
+import com.amplifyframework.storage.s3.transfer.TransferListener
+import com.amplifyframework.storage.s3.transfer.TransferObserver
+import java.io.File
+import java.util.UUID
+import java.util.concurrent.ExecutorService
+
+/**
+ * An operation to upload a file from AWS S3.
+ */
+internal class AWSS3StoragePathUploadFileOperation @JvmOverloads internal constructor(
```

Comment:
Do we need `@JvmOverloads` on these operation constructors? Seems like the plugin is always supplying all values.

Reply:
Will remove. This was from copy paste

Reply:
Will remove on InputStream operation as well.